subsys/nrf_security/src/drivers/cracen/sicrypto/src/iksig.c

```diff
@@ -114,17 +115,12 @@ static int finish_ecdsa_ik_sign(struct sitask *t, struct siwq *wq)
 	return exit_ikg(t);
 }
 
-static int start_ecdsa_ik_sign(struct sitask *t, struct siwq *wq)
+static int start_ecdsa_ik_sign(struct sitask *t)
 {
 	struct sx_pk_acq_req pkreq;
 	struct sx_pk_inops_ik_ecdsa_sign inputs;
 	int opsz;
 	size_t digestsz = sx_hash_get_digestsz(&t->u.h);
```

Comment:
Getting digest size from this hash context won't work when calling `psa_sign_hash()`. This is because the hash context is never created since `sx_hash_create()` is not needed and therefore not called. Changing this line to:
`size_t digestsz = sx_hash_get_alg_digestsz(t->params.ik.privkey->hashalg);`
makes my tests pass